fix(queue): Add expiration check for artifacts (#7412)
taskcluster/taskcluster: changelog/issue-4086.md

audience: users
level: patch
reference: issue 4086
---

> Add check for expiration of artifact in getArtifact API

Reviewer: ```suggestion
`queue.getArtifact()` checks artifact for expiration and returns `ResourceExpired - 410` in such cases
```

Author: fixed 👍
